os/kernel/log_dump/log_dump.c

```diff
@@ -157,7 +157,8 @@ int log_dump_read_wake(void)
 		return ERROR;
 	}
 
-	int	msg_compress = true;
+	int msg_compress = true;
+	writesize = CONFIG_LOG_DUMP_CHUNK_SIZE; /* reset writesize before compression */
```

Comment:
I don't know why this is necessary. Is it changed by the result of the compress_read function even this is initialized as a global variable?

Reply:
As we talked in the chat, we should re-organize the compression API. I think user think it's much difficult to use.